### ISSUE TITLE
bug(ctm): ctm_tensor_c4v converges to a different environment than ctm_tensor — energy off by 4.6e-3 (D=2) to 1.8e-2 (D=4)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#760]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

`ctm_tensor_c4v` and `ctm_tensor` converge to **different** environments for the
same C4v-symmetric iPEPS tensor, and the resulting energies disagree by an
amount that does **not** shrink with χ.

Two convergent approximations of the same contraction must agree in the χ→∞
limit. These don't, so one of them is wrong — and the evidence says it is the
C4v path.

## Evidence

Heisenberg ground states obtained with `optimize_gs_ad` (`unit_cell="1x1"`,
`gs_c4v=True`, sublattice-rotated gate), then contracted two ways with the
**same** observable (`compute_energy_ctm_tensor`) and the **same** gate:

```
D=2 chi=8 : ctm_tensor_c4v=-0.6644886042  ctm_tensor=-0.6599411911  diff=-4.55e-03
D=4 chi=16: ctm_tensor_c4v=-0.6798405407  ctm_tensor=-0.6614978140  diff=-1.83e-02
```

`ctm_tensor` is corroborated three ways and `ctm_tensor_c4v` is not:

* it is **χ-converged to ten digits** on the D=2 state — `-0.6599411913` at
  χ=16, 32 *and* 64;
* it reproduces the optimiser's own reported `E_opt` to 8 digits
  (`-0.66149781` at D=4, `-0.65994119` at D=2);
* it matches the published D=2 C4v iPEPS Heisenberg value ≈ `-0.6602`.

Meanwhile the C4v path drifts the *wrong way* with χ, at D=4 passing straight
through the exact bound (`-0.669437`/site):

```
 chi        E_c4v
   8   -0.6782197088
  16   -0.6798405407
  24   -0.6811723676
  32   -0.6849908793
  48   -0.6857088482
  64   -0.6870429011
```

Finite-χ CTM energies are legitimately non-variational (#511), so dipping below
the bound is not by itself proof. The indictment is that **the gap does not
close**: at D=2 the C4v sweep reports its own residual at 4e-13 (converged) and
`ctm_tensor` is converged to 1e-10, yet they sit 4.8e-3 apart at χ=8 and
4.8e-3 apart at χ=64.

## The state really is C4v

Ruling out "the input isn't symmetric, so replicating one corner is invalid":

```
production CTM corner spectra, D=4 chi=16 (normalised):
  C1: [1, 0.37816, 0.339629, 0.297606, 0.280016, 0.265494]
  C2/C3/C4 identical;  max |C_i - C1| = 1.18e-12
site tensor C4v asymmetry ‖sym(A) - A‖ = 1.8e-15
```

All four corners agree to 1e-12, so `_c4v_to_full_env`'s premise holds. The two
sweeps nonetheless reach different fixed points:

```
D=2 chi=8
  production corner spectrum: [1, 0.175649, 0.154052, 0.026174, 0.005416, 0.003658]
  _c4v_sweep corner spectrum: [1, 0.029928, 0.022471, 0.000644, 9.75e-08, 5.14e-08]
  _c4v_sweep meta: converged=True, 63 iters, residual 4.0e-13
```

(The corner objects may not be strictly comparable between schemes, so treat
the spectra as a pointer, not as the proof — the energy disagreement above is
the load-bearing evidence.)

## Second symptom, same cause

The C4v corner spectrum has fallen to ~1e-8 by its fifth eigenvalue where the
standard CTM is still at 5e-3. Consequently `_c4v_sweep` **stagnates for χ ≳ 16
at D=2** — it truncates inside numerical noise, so the projector picks different
directions every sweep and never converges:

```
D=2, _c4v_sweep, conv_tol=1e-12
  chi=12: converged, 24 iters, residual 7.9e-13
  chi=16: NOT converged after 3000 iters, residual 1.4e-05
  chi=24: NOT converged after 3000 iters, residual 5.4e-09
```

`ctm_tensor` converges on that identical tensor at χ=16/32/64 without
difficulty. Raising `max_iter` 300→3000 makes χ=16 *worse* (4.2e-06 → 1.4e-05),
so this is stagnation, not slow convergence.

## Blast radius

`_c4v_sweep` (`src/tenax/algorithms/_ctm_tensor_c4v.py:37`) backs three callers:

1. **`ctm_tensor_c4v`** — public API, exported in `tenax/__init__.py` `__all__`.
2. **`_ctm_tensor_c4v_reference_ad`** — the reference-C4v path in
   `optimize_gs_ad` (`_use_reference_c4v_path`).
3. **`_ctm_c4v_root_implicit`** — #715 Phase 0, which imports `_c4v_sweep` and
   `_c4v_to_full_env` directly and therefore inherits the defect wholesale.

Worth stressing for #715: the root-implicit machinery is **not** implicated.
Its characteristic-equation residual, adjoint solve and gradient are all
consistent with the environment they are given; that environment is wrong
upstream. The `‖F(y*)‖` check in fact *detects* the χ≥16 stagnation correctly.

## Not the cause

`_c4v_sweep` step 5 builds the new corner as `half @ conj(half).T` where
`half = P†·Cg` already has corner shape `(chi_new, chi)` — i.e. it stores the
projected *density matrix* rather than the projected corner, which predicts a
squared spectrum and roughly matches the numbers above. **Tested and refuted:**
patching step 5 to keep the linear corner moves the energy only in the 6th
decimal and leaves the discrepancy intact.

```
 chi         E_prod      E_c4v_orig    E_c4v_linear
   8   -0.6599411911   -0.6644886042   -0.6644886039
  16   -0.6599411913   -0.6647206463   -0.6647263370
  32   -0.6599411913   -0.6647214764   -0.6647029846
```

A better lead: the corner is grown as `Cg = C·T` only. The double-layer tensor
`a` is absorbed into the *edge* (`Tg = T·a`) but never into the corner, whereas
standard CTMRG enlarges the corner as `C·T·T·a`. Untested — flagging it rather
than asserting it.

## Test gap

Nothing in the suite compares a C4v environment against an independent one.
The #715 Phase 0 tests are FD-vs-AD parity, energy-only-vs-energy-and-grad
self-consistency, and residual/structural checks — every one of them passes
unchanged when the underlying functional is wrong, because they differentiate
or cross-check *the same* functional.

Suggested regression: on a C4v-symmetric state, assert
`compute_energy_ctm_tensor(A, ctm_tensor_c4v(A, chi), gate)` agrees with the
`ctm_tensor` result to CTM tolerance. That single assertion fails today at
4.6e-03 and would have caught this.

## Reproduction

```python
import jax, jax.numpy as jnp
jax.config.update("jax_enable_x64", True)
from tenax import ctm_tensor_c4v, sublattice_rotate_gate, iPEPSConfig, optimize_gs_ad
from tenax.algorithms._ctm_tensor_convergence import ctm_tensor
from tenax.algorithms._ctm_tensor_energy import compute_energy_ctm_tensor
from tenax.algorithms.ipeps import _wrap_as_dense_tensor
from tenax.algorithms.ipeps_config import CTMConfig

Sz = 0.5*jnp.array([[1.,0.],[0.,-1.]]); Sp = jnp.array([[0.,1.],[0.,0.]]); Sm = Sp.T
gate = (jnp.kron(Sz,Sz) + 0.5*jnp.kron(Sp,Sm) + 0.5*jnp.kron(Sm,Sp)).reshape(2,2,2,2)
gate_rot = sublattice_rotate_gate(gate)

D = 2
A0 = _wrap_as_dense_tensor(jax.random.normal(jax.random.PRNGKey(42), (D,D,D,D,2)))
cfg = iPEPSConfig(max_bond_dim=D, ctm=CTMConfig(chi=8, max_iter=100, conv_tol=1e-10),
                  unit_cell="1x1", gs_c4v=True, su_init=False, gs_num_steps=10)
A, _env, E_opt = optimize_gs_ad(gate_rot, A0, cfg)

e_c4v = float(compute_energy_ctm_tensor(A, ctm_tensor_c4v(A, chi=8, max_iter=300,
                                                          conv_tol=1e-12), gate_rot))
env, _ = ctm_tensor(A, 8, max_iter=300, conv_tol=1e-12, projector_method="eigh")
e_std = float(compute_energy_ctm_tensor(A, env, gate_rot))
print(e_c4v, e_std, e_c4v - e_std, float(E_opt))
# -0.6644886042  -0.6599411911  -4.55e-03   (E_opt -0.65994119)
```

Environment: A100, jax x64, branch `fix/715-root-residual-hard-failure` at
parity with `main` (`4783309`).
